integrations/object_store/src/store.rs

```diff
@@ -412,6 +417,74 @@ impl ObjectStore for OpendalStore {
     }
 }
 
+/// `MultipartUpload`'s impl based on `Writer` in opendal
+struct OpendalUpload {
+    // Refer to the fs impl in `object_store`:
+    // https://github.com/apache/arrow-rs/blob/a35214f92ad7c3bce19875bb091cb776447aa49e/object_store/src/local.rs#L715
+    // IO exists during locking, so use tokio::sync::Mutex here
+    writer: Arc<Mutex<Writer>>,
```

Comment:
Sorry for not take deep research on this API first.

`Writer` handles the write internally, all data must be written in order. Using a lock here is incorrect. `Writer` needs to make some changes to make it's possible to implement `MultipartUpload`.

Let me take a look first.

Reply:
Hi, @Rachelint. I'm currently working on a fix for this; let's mark this PR as a draft for now. Thanks for your patience.

Reply:
> Hi, @Rachelint. I'm currently working on a fix for this; let's mark this PR as a draft for now. Thanks for your patience.

Got it. Somethings I can help to fix? I am willing to do more and understand the project better.

Reply:
> Got it. Somethings I can help to fix? I am willing to do more and understand the project better.

Hi, I'm still working on how to address this. Would you like to look at other issues first? Thanks for your understanding!